packages/create-catalyst/src/utils/reset-branch-to-ref.ts

```diff
@@ -0,0 +1,36 @@
+import { execSync } from 'node:child_process';
+
+import { checkoutRef } from './checkout-ref';
+import { isExecException } from './is-exec-exception';
+
+export function resetBranchToRef(repoDir: string, targetBranch: string, ref: string): void {
+  try {
+    // Checkout the target branch
+    const checkoutSuccess = checkoutRef(repoDir, targetBranch);
+
+    // Reset current branch to the specified ref
+    if (checkoutSuccess) {
+      execSync(`git reset --hard ${ref}`, {
+        cwd: repoDir,
+        stdio: 'inherit',
+        encoding: 'utf8',
+      });
+      console.log(`Reset ${targetBranch} to ${ref} successfully.`);
+    }
```

Comment:
Can we rely on errors being thrown instead of boolean conditions? If `checkoutSuccess` is `false`, we essentially do "nothing".

Reply:
Do you mean checking the boolean condition of `checkoutSuccess`? Right now, `checkoutRef` function handles errors gracefully, meaning it doesn't surface it's errors to a caller. Because of that, I don't know how I would know if it ran successfully or not?

Right now, if `checkoutSuccess` is `false` the error handling and logging has already been handled in `checkoutRef`.

Does that address your question?

Reply:
Ummm... so to me CLI application logic flow should be handled by errors being thrown instead of conditional statements like you have currently. It allows us to surface errors and handle them through the catch statement, rather than `if x, do y, else noop`.

I think it should be something like this:
```ts
try {
  checkoutRef(...);

  execSync(...);
} catch (error) {
  if (error instanceof CheckoutRefError) {
    // handle fallback logic
  }

  // ...
}
```

Reply:
Yeah I understand that. The way things are written already, this would mean reworking `checkoutRef` to throw the errors that it's already handling? Then, we have to handle that error every time it's called, which is only once for now and potentially replicating logging for it?